en/docs/learn/design-api/publish-api/publish-an-api-with-artifact-synchronization.md

```diff
@@ -0,0 +1,123 @@
+ gateway's label will be sorted. 
+4. Then it will fetch the artifacts associated with the API from the storage (Database or Github) and load it to the
+ memory.
+
+
+There will be an extension in the publisher profile to store the synapse artifacts in a persistence storage. The default implementation uses the API Manager Database itself. Once the API is Published, Edited, or removed, an event will be sent to Traffic Manager using Event Notifiers with API Name, UUID, and the gateway label for the API. 
+
+
+###API Gateway at the startup
+
+
+  [![]({{base_path}}/assets/img/learn/gateway-startup.png)]({{base_path}}/assets/img/learn/gateway-startup.png)
+  
+At startup, the gateway will look on the APIs with labels which it is subscribed to, in the extension, and fetch the synapse artifacts of those APIs. Those synapse artifacts will get deployed in the gateway.
+
+Gateways are subscribed to the traffic manager. There is an extension in the gateway to get the synapse artifacts and deploy them in the memory. Gateways can subscribe to multiple labels.
+
+###Configuration Related to Artifact Synchronizer
+
+You need to configure Gateway and Publisher node as given below to save artifacts and retrieve artifacts through
+corresponding extensions
+
+##Publisher Profile 
+
+Update `<API-M_HOME>/repository/conf/deployment.toml` file as follows, and change the default values according to your
+ need. 
+
+
+```
+[apim.sync_runtime_artifacts.publisher]
+artifact_saver = "DBSaver"
+publish_directly_to_gateway = "true"
+```
+
+ - Through artifact_saver we can specify the extension point. The default is `DBSaver` where the artifacts are saved in
+ the database.
+ - If `publish_directly_to_gateway = true` then the artifacts will be published to the gateway directly. If
+ `publish_directly_to_gateway = false` then the published API details will be notified to TM through events.
+ - If we have **apim.sync_runtime_artifacts.publisher** config header then all the artifacts will be saved in the
+  extension
+ point.
+
+
+##Gateway Profile Configuration
+
+Update `<API-M_HOME>/repository/conf/deployment.toml` file as follows, and change the default values according to your
+ need. 
+
+```
+[apim.sync_runtime_artifacts.gateway]
+gateway_labels =["Production and Sandbox","Label1","Label2"]
+artifact_retriever = "DBRetriever"
+deployment_retry_duartion = 15000
+data_retrieval_mode = "sync"
+save_artifacts_locally = false
+```
+
+
+ - Through artifact_retriever we can specify the extension point. The default is `DBRetriever` where the artifacts are
+  pulled from the database.
+  
+ - In gateway_labels we can specify the labels which the gateway is going to subscribe to. Only the APIs with these
+  labels will be pulled from the extension point and deployed.
+  
+ - We can specify the retry duration in milliseconds to deploy artifacts if there is a failure in pulling them from the
```

Comment:
Does this still happen after we agreed to limit the retries to a maximum count.

Reply:
We are limiting the synchronous deployment retry attempts . Not the Asynchronous deployment . I have added this part in https://github.com/wso2/docs-apim/pull/1547/commits/df6a2995747f9c1da974a91b5762497536cd947c#diff-8af59c32b66992ca294e395cc8e7f413R95